### ISSUE TITLE
Bug - DBUpdate-to-6.pl wrong psql syntax

### ISSUE THREAD

**Issue by stefan927** [znuny/Znuny#262]
### Environment

- OS:           CentOS 7
- Browser:      Safari
- Znuny version: 6.0.44
- Perl: perl5 (revision 5 version 16 subversion 3)
- PostgreSQL: 10

### Expected behaviour

Successfull migration from OTRS 5.0.42 to Znuny 6.0.44

### Actual behaviour

```
scripts/DBUpdate-to-6.pl

 Migration started ...

 Checking requirements ...

    Requirement check for: Check framework version ...
    Requirement check for: Check required Perl version ...
    Requirement check for: Check required database version ...
    Requirement check for: Check database charset ...
    Requirement check for: Check required Perl modules ...
    Requirement check for: Check if database has been backed up ...

        Did you backup the database? [Y]es/[N]o: Y

    Requirement check for: Migrate PostMaster pre-filters that execute system commands ...
    Requirement check for: Migrate dashboard widgets that execute system commands ...
    Requirement check for: Upgrade database structure ...

Error: The following sequences with possible wrong names have been found. Please rename them manually.
ticket_link_ticket_id_master_seq
ticket_link_ticket_id_slave_seq
pop3_account_id_seq
standard_response_attachment_id_seq
standard_response_id_seq
system_user_id_seq
    Requirement check for: Migrating time zone configuration ...
    Requirement check for: Create entries in new article table ...

        Found 10 orphaned entries in time_accounting table ...
        Do you want to automatically delete the entries from the database now? [Y]es/[N]o: Y
[Mon Jun 27 13:19:05 2022] DBUpdate-to-6.pl: DBD::Pg::db do failed: ERROR:  syntax error at or near "LEFT"
[Mon Jun 27 13:19:05 2022] DBUpdate-to-6.pl: LINE 1: DELETE FROM time_accounting LEFT JOIN article ref ON tab.art...
[Mon Jun 27 13:19:05 2022] DBUpdate-to-6.pl:                                     ^ at /opt/znuny-6.0.44/Kernel/System/DB.pm line 471, <> line 4.
ERROR: OTRS-otrs.Console.pl-Maint::Database::Check-10 Perl: 5.16.3 OS: linux Time: Mon Jun 27 11:19:05 2022

 Message: ERROR:  syntax error at or near "LEFT"
LINE 1: DELETE FROM time_accounting LEFT JOIN article ref ON tab.art...
                                    ^, SQL: 'DELETE FROM time_accounting LEFT JOIN article ref ON tab.article_id = ref.id WHERE tab.article_id IS NOT NULL AND ref.id IS NULL'

 Traceback (56785):
   Module: scripts::DBUpdateTo6::MigrateArticleData::CheckPreviousRequirement Line: 496
   Module: scripts::DBUpdateTo6::_ExecuteComponent Line: 162
   Module: scripts::DBUpdateTo6::Run Line: 68
   Module: scripts/DBUpdate-to-6.pl Line: 88




 Not possible to complete migration, check previous messages for more information.
```

### How to reproduce

1. have a working otrs 5.0.42 installation with psql 10 as db backend
2. execute migration script to znuny 6
3. See error

### Additional information

according to the psql documentation it's not possible to execute DELETE FROM xxxx together with a LEFT JOIN.
i'm in no means a database pro but it looks like this is really a sql syntax error in the psql world.
couldn't find any working solution by googling around :(

**Comment by hanneshal** [comment]
Hi, 

thanks for the report. We dig into this. 

To resolve your migration:

```SQL
SELECT * FROM time_accounting 
LEFT JOIN article ref ON tab.article_id = ref.id WHERE tab.article_id IS NOT NULL AND ref.id IS NULL
```

This should provide you with the 10 "orphaned" entries (ids) that were found.

**If there are exactly 10 items found**, you can then delete them using a sub query or enter the Ids manually.

```SQL
DELETE FROM time_accounting WHERE id IN (SELECT * FROM time_accounting 
LEFT JOIN article ref ON tab.article_id = ref.id WHERE tab.article_id IS NOT NULL AND ref.id IS NULL)
```
Regards

**Comment by stefan927** [comment]
Hi, thx for the help :) 

i ran from error to error now, and i used the following queries in order, they are listed.
(maybe some are not right, but at least i'm now in an error state, where we have some inconsistencies in the database, i need to resolve step by step as well):

```
DELETE FROM time_accounting WHERE id IN (SELECT time_accounting.id FROM time_accounting LEFT JOIN article ref ON article_id = ref.id WHERE article_id IS NOT NULL AND ref.id IS NULL);

DELETE FROM article_flag WHERE article_id IN (SELECT article_id FROM article_flag LEFT JOIN article ref ON article_id = ref.id WHERE article_id IS NOT NULL AND ref.id IS NULL);

DELETE FROM article_plain WHERE id IN (SELECT article_plain.id FROM article_plain LEFT JOIN article ref ON article_id = ref.id WHERE article_id IS NOT NULL AND ref.id IS NULL);

DELETE FROM ticket_history WHERE id IN (SELECT tab.id FROM ticket_history tab LEFT JOIN article ref ON tab.article_id = ref.id WHERE tab.article_id IS NOT NULL AND tab.article_id != 0 AND ref.id IS NULL);

DELETE FROM article WHERE id IN (SELECT tab.id FROM article tab LEFT JOIN ticket ref ON tab.ticket_id = ref.id LEFT JOIN article_flag sub ON tab.id = sub.article_id WHERE tab.ticket_id IS NOT NULL AND ref.id IS NULL);

DELETE FROM article_attachment WHERE article_id IN (SELECT article_id FROM article_attachment LEFT JOIN article ref ON article_id = ref.id WHERE article_id IS NOT NULL AND ref.id IS NULL);

DELETE FROM article_plain WHERE id IN (SELECT article_plain.id FROM article_plain LEFT JOIN article ref ON article_id = ref.id WHERE article_id IS NOT NULL AND ref.id IS NULL);

DELETE FROM ticket_history WHERE id IN (SELECT tab.id FROM ticket_history tab LEFT JOIN article ref ON tab.article_id = ref.id WHERE tab.article_id IS NOT NULL AND tab.article_id != 0 AND ref.id IS NULL);
```

please correct, if those are garbage. i still have a full backup + snapshot, so no big deal if the whole thing blow's up.

kinda offtopic:
those inconsistency's in the db are pretty sure, from all the migrations this otrs instance has behind it...
(or my delete's did something unexpected)
if i digged right, it dates back to otrs-2.4 - arcticle.id is at 21,5 mio.

basically now i'm getting errors for foreign key constraint violations, 
which i can confirm by comparing the tables, pretty ugly uaaah :(

**Comment by rkaldung** [comment]
Since a workaround exists, we won't fix this. There are now release possible for end-of-life versions.
